server/sonar-web/src/main/js/components/SourceViewer/helpers/highlight.ts

```diff
@@ -17,38 +17,37 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-// @flow
-import escapeHtml from 'escape-html';
 import { uniq } from 'lodash';
+import { LinearIssueLocation } from '../../../app/types';
 
-/*::
-export type Token = { className: string, markers: Array<number>, text: string };
-*/
-/*::
-export type Tokens = Array<Token>; */
+export interface Token {
+  className: string;
+  markers: number[];
+  text: string;
+}
 
 const ISSUE_LOCATION_CLASS = 'source-line-code-issue';
 
-export function splitByTokens(code /*: string */, rootClassName /*: string */ = '') /*: Tokens */ {
+export function splitByTokens(code: string, rootClassName = ''): Token[] {
   const container = document.createElement('div');
-  let tokens = [];
+  let tokens: Token[] = [];
   container.innerHTML = code;
-  [].forEach.call(container.childNodes, node => {
+  [].forEach.call(container.childNodes, (node: Element) => {
     if (node.nodeType === 1) {
       // ELEMENT NODE
       const fullClassName = rootClassName ? rootClassName + ' ' + node.className : node.className;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Use a template literal instead of concatenating with a string literal. ([typescript:S3512](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3512))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fpopups&issues=AWHb6qbLw6XGHKvPuQTU&open=AWHb6qbLw6XGHKvPuQTU)

[ISSUE_KEY]: AWHb6qbLw6XGHKvPuQTU